subsys/bluetooth/shell/bt.c

```diff
@@ -1122,6 +1122,38 @@ static int cmd_bondable(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static void bond_info(const struct bt_bond_info *info, void *user_data)
+{
+	int *bond_count = (int *)user_data;
```

Comment:
minor: remove the explicit `(int *)` typecast. The whole idea with void pointers is that they can be implicitly cast to any other pointer type.